incidents/src/test/java/fr/openent/incidents/service/impl/IncidentsTypeServiceTest.java

```diff
@@ -0,0 +1,142 @@
+            async.complete();
+        });
+
+        Whitebox.invokeMethod(this.incidentsService, "fetchUsedIncidentsType", "structureId", (Handler) e -> {
+
+        });
+    }
+
+    @Test
+    public void testCreate(TestContext ctx) throws Exception {
+        Async async = ctx.async();
+
+        String expectedQuery = "INSERT INTO null.incident_type (structure_id, label, hidden) VALUES (?, ?, false) RETURNING id";
+        JsonArray expectedParams = new JsonArray().add("structureId").add("label");
+        JsonObject incidentTypeBody = new JsonObject().put("structureId", "structureId").put("label", "label");
+
+        vertx.eventBus().consumer("fr.openent.incidents", message -> {
+            JsonObject body = (JsonObject) message.body();
+            ctx.assertEquals("prepared", body.getString("action"));
+            ctx.assertEquals(expectedQuery, body.getString("statement"));
+            ctx.assertEquals(expectedParams.toString(), body.getJsonArray("values").toString());
+            async.complete();
+        });
+
+        Whitebox.invokeMethod(this.incidentsService, "create", incidentTypeBody, (Handler) e -> {
+
+        });
+    }
+
+    @Test
+    public void testPut(TestContext ctx) throws Exception {
+        Async async = ctx.async();
+
+        String expectedQuery = "UPDATE null.incident_type SET label = ?, hidden = ? WHERE id = ? RETURNING id";
+        JsonArray expectedParams = new JsonArray().add("label").add(true).add(3);
+        JsonObject incidentTypeBody = new JsonObject().put("label", "label").put("hidden", true).put("id", 3);
+
+        vertx.eventBus().consumer("fr.openent.incidents", message -> {
+            JsonObject body = (JsonObject) message.body();
+            ctx.assertEquals("prepared", body.getString("action"));
+            ctx.assertEquals(expectedQuery, body.getString("statement"));
+            ctx.assertEquals(expectedParams.toString(), body.getJsonArray("values").toString());
+            async.complete();
+        });
+
+        Whitebox.invokeMethod(this.incidentsService, "put", incidentTypeBody, (Handler) e -> {
+
+        });
+    }
+
+    @Test
+    public void testDelete(TestContext ctx) throws Exception {
+        Async async = ctx.async();
+
+        String expectedQuery = "DELETE FROM null.incident_type WHERE id = ? RETURNING id as id_deleted";
+        JsonArray expectedParams = new JsonArray().add(3);
+
+        vertx.eventBus().consumer("fr.openent.incidents", message -> {
+            JsonObject body = (JsonObject) message.body();
+            ctx.assertEquals("prepared", body.getString("action"));
+            ctx.assertEquals(expectedQuery, body.getString("statement"));
+            ctx.assertEquals(expectedParams.toString(), body.getJsonArray("values").toString());
+            async.complete();
+        });
+
+        Whitebox.invokeMethod(this.incidentsService, "delete", 3, (Handler) e -> {
+
```

Comment:
Whitebox.invokeMethod(this.incidentsService, "fetchUsedIncidentsType", "structureId", null)